src/components/set-sounds-invisible.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * Sets all non-positional audio children to not be visible so their world matrices are not updated.
+ */
+AFRAME.registerComponent("set-sounds-invisible", {
+  schema: {
+    withinDistance: { type: "number" }
+  },
+
+  init() {
+    for (const component of Object.values(this.el.components)) {
+      if (component.attrName !== "sound" && !component.attrName.startsWith("sound__")) continue;
+      if (component.data.positional) continue;
```

Comment:
reflecting on it i like it better breaking these up instead of doing the ((x && y) || z). the first guard clause is filtering the components the second is filtering the positional ones